### PR TITLE
[IOTDB-405] cache results in the client and makes the display look nice.

### PR DESCRIPTION
**Pull request by Genius-pig**
cache results in the client and makes the display look nice.
https://issues.apache.org/jira/projects/IOTDB/issues/IOTDB-405

### CODE REVIEW

File: client/src/main/java/org/apache/iotdb/client/AbstractClient.java [hunk 1]
```diff
@@ -82,89 +82,29 @@
   static final String TIMESTAMP_STR = "Time";
   static final int ISO_DATETIME_LEN = 35;
   private static final String IMPORT_CMD = "import";
-  private static final String NEED_NOT_TO_PRINT_TIMESTAMP = "AGGREGATION";
   private static final String DEFAULT_TIME_FORMAT = "default";
   private static String timeFormat = DEFAULT_TIME_FORMAT;
   static int maxPrintRowCount = 100000;
```
Comment (qiaojialin) [inline client/src/main/java/org/apache/iotdb/client/AbstractClient.java]:
```suggestion
  static int maxPrintRowCount = 1000;
```

File: client/src/main/java/org/apache/iotdb/client/AbstractClient.java [hunk 2]
```diff
@@ -82,89 +82,29 @@
...
   private static int fetchSize = 10000;
```
Comment (qiaojialin) [inline client/src/main/java/org/apache/iotdb/client/AbstractClient.java]:
```suggestion
  private static int fetchSize = 1000;
```

File: client/src/main/java/org/apache/iotdb/client/AbstractClient.java [hunk 3]
```diff
@@ -805,34 +503,100 @@ private static void importCmd(String specialCmd, String cmd, IoTDBConnection con
...
     println(String.format("It costs %.3fs", costTime / 1000.0));
   }
 
+  private static List<List<String>> cacheResult(ResultSet resultSet, List<Integer> maxSizeList,
+      int columnLength, ResultSetMetaData resultSetMetaData, ZoneId zoneId)
+      throws SQLException {
+    boolean printTimestamp = !((IoTDBQueryResultSet) resultSet).isIgnoreTimeStamp();
+    List<List<String>> lists = new ArrayList<>(columnLength);
+    for(int i = 1; i <= columnLength; i++) {
+      List<String> list = new ArrayList<>(30);
```
Comment (qiaojialin) [inline client/src/main/java/org/apache/iotdb/client/AbstractClient.java]: change this 30 to a field of this class

File: client/src/main/java/org/apache/iotdb/client/AbstractClient.java [hunk 4]
```diff
@@ -805,34 +503,100 @@ private static void importCmd(String specialCmd, String cmd, IoTDBConnection con
...
+      }
+      for(int i = 1; i <= columnLength; i++) {
+        String tmp;
+        if(printTimestamp && i == 1) {
+          tmp = formatDatetime(resultSet.getLong(TIMESTAMP_STR), zoneId);
+        } else {
+          tmp = resultSet.getString(i);
+        }
+        lists.get(i-1).add(tmp);
+        if(maxSizeList.get(i-1) < tmp.length()) {
```
Comment (qiaojialin) [inline client/src/main/java/org/apache/iotdb/client/AbstractClient.java]: format

File: client/src/main/java/org/apache/iotdb/client/AbstractClient.java [hunk 5]
```diff
@@ -805,34 +503,100 @@ private static void importCmd(String specialCmd, String cmd, IoTDBConnection con
...
       }
     } catch (Exception e) {
       println("Msg: " + e.getMessage());
-      handleException(e);
     }
     long costTime = System.currentTimeMillis() - startTime;
     println(String.format("It costs %.3fs", costTime / 1000.0));
   }
 
+  private static List<List<String>> cacheResult(ResultSet resultSet, List<Integer> maxSizeList,
```
Comment (qiaojialin) [inline client/src/main/java/org/apache/iotdb/client/AbstractClient.java]: add some javadoc

File: client/src/main/java/org/apache/iotdb/client/AbstractClient.java [hunk 6]
```diff
@@ -805,34 +503,100 @@ private static void importCmd(String specialCmd, String cmd, IoTDBConnection con
...
+          ResultSetMetaData resultSetMetaData = resultSet.getMetaData();
+          int columnLength = resultSetMetaData.getColumnCount();
+          List<Integer> maxSizeList = new ArrayList<>(columnLength);
+          List<List<String>> lists = cacheResult(resultSet, maxSizeList, columnLength,
+              resultSetMetaData, zoneId);
+          output(lists, maxSizeList);
         }
       }
     } catch (Exception e) {
       println("Msg: " + e.getMessage());
```
Comment (qiaojialin) [inline client/src/main/java/org/apache/iotdb/client/AbstractClient.java]: There is a null pointer exception, I can not print result in my client. Try to fix it...

**Comment by qiaojialin** [comment]
My test case is:

SET STORAGE GROUP TO root.turbine;
create timeseries root.turbine.d1.s0 with datatype=DOUBLE, encoding=GORILLA, compression=SNAPPY;
create timeseries root.turbine.d2.s0 with datatype=DOUBLE, encoding=GORILLA, compression=SNAPPY;
insert into root.turbine.d1(time,s0) values(4,4);
select * from root;

File: client/src/main/java/org/apache/iotdb/client/AbstractClient.java (hunk 4 above)
Comment (Genius-pig) [reply client/src/main/java/org/apache/iotdb/client/AbstractClient.java]: format what?

File: client/src/main/java/org/apache/iotdb/client/AbstractClient.java (hunk 6 above)
Comment (Genius-pig) [reply client/src/main/java/org/apache/iotdb/client/AbstractClient.java]: fixed.

File: client/src/main/java/org/apache/iotdb/client/AbstractClient.java (hunk 5 above)
Comment (Genius-pig) [reply client/src/main/java/org/apache/iotdb/client/AbstractClient.java]: fixed

File: client/src/main/java/org/apache/iotdb/client/AbstractClient.java (hunk 4 above)
Comment (qiaojialin) [reply client/src/main/java/org/apache/iotdb/client/AbstractClient.java]: oh, ignore me

**Pull request merged**